pkg/users/account-create-dialog.js

```diff
@@ -220,57 +297,74 @@ export function account_create_dialog(accounts) {
-                .catch(ex => {
-                    errs.password = (ex.message || ex.toString()).replaceAll("\n", " ");
-                })
+        if (!state.home_dir)
+            errs.home_dir = _("No home directory specified");
+
+        const promises = [
+            password_quality(password, force_weak)
+                    .catch(ex => {
+                        errs.password = (ex.message || ex.toString()).replaceAll("\n", " ");
+                    })
+        ];
+        if (!force_home) {
+            promises.push(
+                validate_home_dir(state.home_dir)
+                        .catch(() => {
+                            errs.home_dir = cockpit.format(_("The home directory $0 already exists. Its ownership will be changed to the new user."), state.home_dir);
+                            state.home_dir_exists = true;
+                        })
+            );
+        }
+
+        return Promise.all(promises)
                 .then(() => {
                     errors = errs;
                     return !has_errors(errs);
                 });
     }
 
-    function create(real_name, user_name, password, locked, force_change) {
-        return cockpit.spawn(["/usr/sbin/useradd", "-D"], { superuser: "require" })
-                .catch(() => "")
-                .then(defaults => {
-                    let shell = null;
-                    defaults.split("\n").forEach(item => {
-                        if (item.indexOf("SHELL=") === 0) {
-                            shell = item.split("=")[1] || "";
-                        }
-                    });
-                    const prog = ["/usr/sbin/useradd", "--create-home", "-s", shell || "/bin/bash"];
-                    if (real_name) {
-                        prog.push('-c');
-                        prog.push(real_name);
+    function create(real_name, user_name, password, locked, force_change, home_dir, force_home) {
+        const prog = ["useradd", "--create-home", "-s", state.shell];
+        if (real_name) {
+            prog.push('-c');
+            prog.push(real_name);
+        }
+
+        if (home_dir) {
+            prog.push('-d');
+            prog.push(home_dir);
+        }
+        prog.push(user_name);
+        return cockpit.spawn(prog, { superuser: "require", err: "message" })
+                .then(() => passwd_change(user_name, password))
+                .then(() => {
+                    if (locked)
+                        return cockpit.spawn([
+                            "usermod",
+                            user_name,
+                            "--lock"
+                        ], { superuser: "require", err: "message" });
+                    if (force_change)
+                        return cockpit.spawn([
+                            "passwd",
+                            "-e",
+                            user_name
+                        ], { superuser: "require", err: "message" });
```

Comment:
These 16 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230314-140716-6b50d398-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)